draft-thomassen-dnsop-multialgo.md

```diff
@@ -157,25 +164,30 @@ The following algorithms are thus INSECURE: 1, 3, 5, 6, 7, 12
 
 ## Signing Requirements
 
- 1. Signers must sign with at least one UNIVERSAL algorithm if any
-    are present in the DS RRset or trust anchor set.  Other
-    signatures are OPTIONAL.
+ 1. Signers must sign with at least one UNIVERSAL algorithm if at
+    least one UNIVERSAL algorithm is present in the DS RRset or trust
+    anchor set.  Other signatures are OPTIONAL.
 
  2. Absent any UNIVERSAL algorithms in the DS RRset or trust anchor
     set, signers MUST sign with all algorithm listed.
 
 ## Validator Requirements
 
  1. When the DS RRset or trust anchor set for a zone includes an
-    unsupported INSECURE algorithm, validators MUST treat the zone as
-    unsigned, even if signed with another supported algorithm.
+    unsupported INSECURE algorithm, validators MUST be willing to treat the
+    zone as unsigned, even if the DS RRset or trust anchor set lists another
+    supported algorithm.  More specifically, unvalidated responses bearing
+    just the unsupported RRSIGs MUST be accepted as valid "insecure" responses.
+    Response RRsets with supported signatures SHOULD still be validated.
```

Comment:
Yes, AD=1 can be set for any validated answer. This is not made invalid by the possibility that some other answer in the past or future might not be validated.  The validated answer can, for example, be relied on in DANE (with TLSA RRs) or SSH (with SSHFP RRs)... The possibility of an unrealised downgrade attack is not relevant for the records that do have valid RRSIGs.

Even today, a domain could go unsigned (DS RRs removed) shortly after a validated answer is received.

Reply:
Sure. The issue is, though, that knowledge of my resolver's algorithm support and inspection of a zone's DS records won't tell me anymore if my resolver will give me a validated answer or not -- even when the DS RRset is constant --, because it now might depend on which RRSIGs the resolver happens to see. Also, an attacker can always take a response with a supported RRSIG and change the algorithm number to the unsupported one, rendering the answer insecure (a rather unexpected possibility). It would be more consistent if it were just insecure.

Elsewhere in the document, we say "that the zone's security status can be established upon inspection of a DS record or TA set". With your suggestion, this would no longer be correct

Plus, it would seem to raise the implementation barrier for resolvers: When I talked to one implementor last week, they told me:
> What I certainly did find very useful in implementation [of the current specs] is the property that you decide the downgrade already when knowing just the DS set.

This seems to be in line with your answer to Q8 in https://mailarchive.ietf.org/arch/msg/dnsop/zwSaDJ-SPd3in0VuD_ncj_vsYsA/.

My take is that if a zone owner uses a "MUST NOT sign" (= INSECURE) algorithm that a validator doesn't support, it's entirely on the zone owner to recover validatability by switching to a permissible algorithm. Until then, it's ok that they get the same security guarantees as in a single-signer setup with only the unsupported algorithm (that is, "insecure" always). It's not for the resolver to bend over to sort out that broken setup.

My impression is that there's a little more to discuss here. To make progress with the document merge, I'll skip over this for now, but will open an issue in the future document's repository to track this aspect.

Reply:
https://github.com/shuque/draft-dnsop-multi-alg-rules/issues/3